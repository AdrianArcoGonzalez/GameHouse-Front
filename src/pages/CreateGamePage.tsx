import CreateGame from "../components/CreateGame/CreateGame";

const CreateGamePage = (): JSX.Element => {
  return (
    <>
      <h2 className="title">Create a Game</h2>
      <CreateGame />
    </>
  );
};

export default CreateGamePage;
