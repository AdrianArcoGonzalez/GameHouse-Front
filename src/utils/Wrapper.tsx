import * as reactRedux from "react-redux";
import { store } from "../store/store";

export interface WrapperProps {
  children: JSX.Element | JSX.Element[];
}

export const Wrapper = ({ children }: WrapperProps): JSX.Element => {
  return <reactRedux.Provider store={store}>{children}</reactRedux.Provider>;
};
