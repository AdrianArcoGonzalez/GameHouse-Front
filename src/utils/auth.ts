import jwt from "jwt-decode";

const decodeToken = (token: string) => {
  const payloadToken: {
    id: string;
    username: string;
    iat: number;
  } = jwt(token);

  const user = {
    token: token,
    id: payloadToken.id,
    username: payloadToken.username,
    isLogged: true,
  };
  return user;
};

export default decodeToken;
