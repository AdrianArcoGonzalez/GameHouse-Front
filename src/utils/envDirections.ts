export const backUrl = process.env.REACT_APP_URL_BACK

export const urlsBack = {
  default: `${backUrl}games/games/`,
  getByOwner: `${backUrl}games/games/my-collection/`,
  getByCategory: `${backUrl}games/games/category/`,

}
