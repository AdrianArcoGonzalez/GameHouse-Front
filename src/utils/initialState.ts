import decodeToken from "./auth";

const emptyUser = {
  id: "",
  username: "",
  token: "",
  isLogged: false,
};
const initialState = () => {
  const token = localStorage.getItem("token") as string;

  if (token) {
    return decodeToken(token);
  }

  return emptyUser;
};

export default initialState;
