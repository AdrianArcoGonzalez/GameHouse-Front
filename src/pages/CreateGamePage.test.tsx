import { Wrapper } from "../utils/Wrapper";
import CreateGamePage from "./CreateGamePage";

import TestRenderer from "react-test-renderer";

describe("Given a createGamePage component", () => {
  describe("When it's instantiated", () => {
    test("Then it should match the screenshoot", async () => {
      const createGamePage = TestRenderer.create(
        <Wrapper>
          <CreateGamePage />
        </Wrapper>
      );

      expect(createGamePage).toMatchSnapshot();
    });
  });
});
