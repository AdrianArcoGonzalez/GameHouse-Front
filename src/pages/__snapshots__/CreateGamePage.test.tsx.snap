// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a createGamePage component When it's instantiated Then it should match the screenshoot 1`] = `
Array [
  <section
    className="sc-gsnTZi cUELPY"
    data-testid="section"
  >
    <img
      alt="backgroundImage"
      className="hero__image"
      height="200px"
      src="/images/addGame.webp"
      width="100vw"
    />
    <div
      className="hero-container"
    >
      <p
        className="hero-container__sentence"
      >
        ”If our lives are already written, it would take a courageos man to chenge the script.” (Alan Wake)
      </p>
    </div>
  </section>,
  <section
    className="sc-dkzDqf mLREY"
  >
    <h2
      className="title"
    >
      Add New Game
    </h2>
    <form
      autoComplete="off"
      className="sc-bczRLJ hyDoqy"
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="input-container"
      >
        <label
          className="form__input-label"
          htmlFor="title"
        >
          Title
        </label>
        <input
          className="form__input-element"
          id="title"
          onChange={[Function]}
          placeholder="Assassins Creed 3"
          type="text"
          value=""
        />
      </div>
      <div
        className="input-container"
      >
        <label
          className="form__input-label"
          htmlFor="company"
        >
          Company
        </label>
        <input
          className="form__input-element"
          id="company"
          onChange={[Function]}
          placeholder="FromSoftware"
          type="text"
          value=""
        />
      </div>
      <div
        className="input-container"
      >
        <label
          className="form__input-label"
          htmlFor="category"
        >
          Category
        </label>
        <select
          className="form__input-element"
          id="category"
          onChange={[Function]}
        >
          <option>
            Adventure
          </option>
          <option>
            Shooter
          </option>
          <option>
            Strategy
          </option>
          <option>
            MOBA
          </option>
          <option>
            Shooter
          </option>
          <option>
            Platforms
          </option>
          <option>
            Simulator
          </option>
        </select>
      </div>
      <div
        className="input-container"
      >
        <label
          className="form__input-label"
          htmlFor="sinopsis"
        >
          Sinopsis
        </label>
        <textarea
          className="form__input-element"
          cols={20}
          id="sinopsis"
          onChange={[Function]}
          placeholder="Add a description of the game"
          rows={10}
        />
      </div>
      <div
        className="input-container"
      >
        <label
          className="form__input-label"
          htmlFor="image"
        >
          Image
        </label>
        <input
          className="form__input-element"
          id="image"
          onChange={[Function]}
          type="file"
          value=""
        />
      </div>
      <button
        className="form-button"
        disabled={true}
        type="submit"
      >
        Create
      </button>
    </form>
  </section>,
]
`;
